code/site/components/com_pages/template/helper/snippet.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+class ComPagesTemplateHelperSnippet extends ComPagesTemplateHelperAbstract
+{
+    private $__snippets = array();
+
+    public function define(string $name, string $snippet, $overwrite = false)
+    {
+        if(!isset($this->__snippets[$name]) || $overwrite) {
+            $this->__snippets[$name] = $snippet;
+        }
+    }
+
+    public function expand(string $name, array $variables = array())
+    {
+        $result = false;
+
+        if(isset($this->__snippets[$name]))
+        {
+            $snippet = $this->__snippets[$name];
+
+            //Use the stream buffer to evaluate the partial
+            $str = "<?php \n return <<<SNIPPET\n$snippet\nSNIPPET;\n";
+
+            $buffer = $this->getObject('filesystem.stream.factory')->createStream('koowa-buffer://temp', 'w+b');
+            $buffer->truncate(0);
+            $buffer->write($str);
```

Comment:
@johanjanssens Does it make sense or is it possible to cache these buffers instead of the snippets? Would save us some processing?

Reply:
I’m happy to work on all these smaller changes, but right now I already lost more then a week on this ‘make html in helpers’ flexible. Starting to feel like micro-optimisations for problems we don’t immediately have.

Reply:
Fair enough. It should be quite fast compared to template rendering anyway so let's roll with it.

Reply:
@ercanozkaya Not giving up that easily. I changed the implementation to use the php template engine. This way we have build in caching out of the box. See: https://github.com/joomlatools/joomlatools-pages/pull/147/commits/9a1acd9451c18cc68a1ab6f17b346d594ef47201